flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1061,6 +1063,20 @@ void markFinished(Map<String, Accumulator<?, ?>> userAccumulators, IOMetrics met
 							IntermediateResultPartition[] allPartitions = finishedPartition
 									.getIntermediateResult().getPartitions();
 
+							PartitionTable<ResourceID> partitionTable = vertex.getExecutionGraph().getPartitionTable();
```

Comment:
We could consider introducing a separate component something like `PartitionTracker` which would also encapsulate `ShuffleMaster` (like `JobAwareShuffleEnvironmentImpl`). It could be responsible for doing what we have in this PR and maybe even register with the `ShuffleMaster` and track `producedPartitions` by execution id as well.

Reply:
yes we could do that. Not sure whether we should do this now or defer this to a later point. Currently we'd just end up wrapping the partition table and mirroring the API.